collectors/modality-probe-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,550 @@
+use err_derive::Error;
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::prelude::*;
+use std::net::SocketAddrV4;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use goblin::elf::Elf;
+
+use modality_probe_debug_collector::{Config, ProbeAddr, Target, Word};
+
+#[derive(Debug, Error)]
+pub enum CLIError {
+    #[error(display = "Address is not valid or too large \"{}\"", _0)]
+    AddressNotValid(String),
+    #[error(display = "Error opening ELF file")]
+    ElfFileError,
+    #[error(display = "Symbols were given but no elf file was specified")]
+    MissingElfFileError,
+    #[error(display = "Given interval not a valid duration: {}", _0)]
+    InvalidInterval(String),
+    #[error(display = "Symbol not found in given ELF file: \"{}\"", _0)]
+    SymbolNotFound(String),
+}
+
+#[derive(Debug, Default, StructOpt)]
+#[structopt(
+    name = "modality-probe-debug-collector",
+    about = "Periodically collects logs from microcontrollers over debug interfaces; outputs them to a file."
+)]
+pub struct CLIOptions {
+    /// Session id to associate with the collected trace data
+    #[structopt(short = "s", long = "session-id", default_value = "0")]
+    session_id: u32,
+
+    /// Specifies 32 bit architecture of target system
+    #[structopt(short = "32", long = "32-bit", conflicts_with = "word-size-64")]
+    word_size_32: bool,
+
+    /// Specifies 32 bit architecture of target system
+    #[structopt(short = "64", long = "64-bit")]
+    word_size_64: bool,
```

Comment:
Can't we parse this out of the elf file header?

Reply:
I suppose if we allow the system architecture to differ from the elf file, this serves a value.

Reply:
Also, as it currently stands, the user does not necessarily need to supply an elf file.